enterprise/backend/src/metabase_enterprise/advanced_permissions/common.clj

```diff
@@ -15,17 +15,25 @@
   "Enterprise version. Returns a boolean whether the current user can write the given field."
   :feature :advanced-permissions
   [instance]
-  (= :yes (data-perms/table-permission-for-user
-           api/*current-user-id*
-           :perms/manage-table-metadata
-           (or (get-in instance [:table :db_id])
-               (database/table-id->database-id (:table_id instance)))
-           (:table_id instance))))
+  (let [db-id (or (get-in instance [:table :db_id])
+                  (database/table-id->database-id (:table_id instance)))]
+    (data-perms/user-has-permission-for-table?
+     api/*current-user-id*
+     :perms/manage-table-metadata
+     :yes
+     db-id
+     (:table_id instance))))
 
 (defenterprise current-user-can-read-schema?
   "Enterprise version. Returns a boolean whether the current user can read the given schema"
   :feature :advanced-permissions
   [db-id schema-name]
+  (data-perms/user-has-permission-for-schema?
+   api/*current-user-id*
+   :perms/manage-table-metadata
+   :yes
+   db-id
+   schema-name)
   (= :yes (data-perms/schema-permission-for-user api/*current-user-id*
```

Comment:
Think this was left in by mistake and can be deleted?

Reply:
Er, line 37 to the right.

Reply:
indeed, good catch, thanks